### PR TITLE
ENH: Using ITKMontage module in the most basic way

### PR DESCRIPTION
**Pull request by dzenanz**
This produces something when pointed at `C:\Misc\BlueQuartz\ITKMontage\test\Input\FlatField\14`, and ParaView 5.6.0 can visualize it.

### CODE REVIEW

File: Documentation/MultiscaleFusionFilters/ITKMontageFromFilesystem.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+MyFilter
```
Comment (thewtex) [inline Documentation/MultiscaleFusionFilters/ITKMontageFromFilesystem.md:1]: Give me a name :-)

File: Documentation/MultiscaleFusionFilters/ITKMontageFromFilesystem.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+MyFilter 
+=============
+
+## Group (Subgroup) ##
+MultiscaleFusion (MultiscaleFusion)
+
+## Description ##
+This **Filter** does the following...
```
Comment (thewtex) [inline Documentation/MultiscaleFusionFilters/ITKMontageFromFilesystem.md:8]: Describe me

File: Documentation/MultiscaleFusionFilters/ITKMontageFromFilesystem.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## Parameters ##
+| Name | Type | Description |
+|------|------|------|
+| Parameter Name | Parameter Type | Description of parameter... |
```
Comment (thewtex) [inline Documentation/MultiscaleFusionFilters/ITKMontageFromFilesystem.md:13]: Describe the file list use

File: Documentation/MultiscaleFusionFilters/ITKMontageFromFilesystem.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+| **Data Container** | Data Container Name | N/A | N/A | Description of object... |
+| **Attribute Matrix** | Attribute Matrix Name | Element/Feature/Ensemble/etc. | N/A | Description of object... |
+| **Element/Feature/Ensemble/etc. Attribute Array** | AttributeArray Name | int32_t/float/etc. | (1)/(3)/etc. | Description of object... |
+
+## Created Objects ##
+| Kind | Default Name | Type | Component Dimensions | Description |
+|------|--------------|-------------|---------|-----|
+| **Data Container** | Data Container Name | N/A | N/A | Description of object... |
+| **Attribute Matrix** | Attribute Matrix Name | Element/Feature/Ensemble/etc. | N/A | Description of object... |
+| **Element/Feature/Ensemble/etc. Attribute Array** | AttributeArray Name | int32_t/float/etc. | (1)/(3)/etc. | Description of object... |
```
Comment (thewtex) [inline Documentation/MultiscaleFusionFilters/ITKMontageFromFilesystem.md:30]: What is the destination montage, the destination transforms

File: Documentation/MultiscaleFusionFilters/ITKMontageFromFilesystem.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+## License & Copyright ##
+
+Please see the description file distributed with this plugin.
```
Comment (thewtex) [inline Documentation/MultiscaleFusionFilters/ITKMontageFromFilesystem.md:34]: Add license and copyright file

File: MultiscaleFusionFilters/ITKMontageFromFilesystem.h [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+
+public:
+  SIMPL_SHARED_POINTERS(ITKMontageFromFilesystem)
+  SIMPL_FILTER_NEW_MACRO(ITKMontageFromFilesystem)
+  SIMPL_TYPE_MACRO_SUPER(ITKMontageFromFilesystem, AbstractFilter)
+
+  ITK_DISALLOW_COPY_AND_ASSIGN(ITKMontageFromFilesystem);
+  ~ITKMontageFromFilesystem() override;
+
+  SIMPL_FILTER_PARAMETER(IntVec3_t, MontageSize)
```
Comment (thewtex) [inline MultiscaleFusionFilters/ITKMontageFromFilesystem.h:44]: Document

File: MultiscaleFusionFilters/ITKMontageFromFilesystem.h [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+  static constexpr unsigned Dimension = 2;
+  using PointType = itk::Point<double, Dimension>;
+  using PositionTableType = std::vector<std::vector<PointType>>;
+  using FilenameTableType = std::vector<std::vector<std::string>>;
+
+  /**
+   * @brief Loads montage input configuration from a TileConfiguration.txt file
+   */
+  void loadTileConfiguration(std::string dirPath, unsigned xSize, unsigned ySize, PositionTableType& pos, FilenameTableType& files);
+
```
Comment (thewtex) [inline MultiscaleFusionFilters/ITKMontageFromFilesystem.h:169]: also test input where there is no tile configuration

Add overlap as a parameter, have a good default

**Review by thewtex — CHANGES_REQUESTED** [review]

**Comment by thewtex** [comment]
I am running into the following build error:

```
In file included from /home/matt/src/SIMPL/Source/SIMPLib/ITK/itkDream3DITransformContainerToTransform.h:87:0,
                 from /home/matt/src/SIMPL/Source/SIMPLib/ITK/Testing/Cxx/itkDream3DITransformContainerToTransformTest.cpp:36,
                 from SIMPLView/SIMPL/SIMPLib/Testing/SIMPLUnitTest.cpp:40:
/home/matt/src/SIMPL/Source/SIMPLib/ITK/itkDream3DITransformContainerToTransform.h: In instantiation of ‘class itk::Dream3DITransformContainerToTransform<float, 3>’:
/home/matt/src/SIMPL/Source/SIMPLib/ITK/Testing/Cxx/itkDream3DITransformContainerToTransformTest.cpp:53:15:   required from here
/home/matt/src/SIMPL/Source/SIMPLib/ITK/itkDream3DITransformContainerToTransform.hxx:89:1: error: ‘void itk::Dream3DITransformContainerToTransform<TParametersValueType, CompositeNDimensions>::VerifyPreconditions() [with TParametersValueType = float; unsigned int CompositeNDimensions = 3]’ marked ‘override’, but does not override
 Dream3DITransformContainerToTransform<TParametersValueType, CompositeNDimensions>
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/matt/src/SIMPL/Source/SIMPLib/ITK/itkDream3DTransformContainerToTransform.h:78:0,
                 from /home/matt/src/SIMPL/Source/SIMPLib/ITK/Testing/Cxx/itkDream3DTransformContainerToTransformTest.cpp:34,
                 from SIMPLView/SIMPL/SIMPLib/Testing/SIMPLUnitTest.cpp:41:
/home/matt/src/SIMPL/Source/SIMPLib/ITK/itkDream3DTransformContainerToTransform.h: In instantiation of ‘class itk::Dream3DTransformContainerToTransform<itk::AffineTransform<float, 3> >’:
/home/matt/src/SIMPL/Source/SIMPLib/ITK/Testing/Cxx/itkDream3DTransformContainerToTransformTest.cpp:53:15:   required from here
/home/matt/src/SIMPL/Source/SIMPLib/ITK/itkDream3DTransformContainerToTransform.hxx:85:1: error: ‘void itk::Dream3DTransformContainerToTransform<ITKTransformType>::VerifyPreconditions() [with ITKTransformType = itk::AffineTransform<float, 3>]’ marked ‘override’, but does not override
 Dream3DTransformContainerToTransform<ITKTransformType>
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/matt/src/SIMPL/Source/SIMPLib/ITK/itkTransformToDream3DTransformContainer.h:78:0,
                 from /home/matt/src/SIMPL/Source/SIMPLib/ITK/Testing/Cxx/itkTransformToDream3DTransformContainerTest.cpp:34,
                 from SIMPLView/SIMPL/SIMPLib/Testing/SIMPLUnitTest.cpp:42:
/home/matt/src/SIMPL/Source/SIMPLib/ITK/itkTransformToDream3DTransformContainer.h: In instantiation of ‘class itk::TransformToDream3DTransformContainer<itk::AffineTransform<float, 3> >’:
/home/matt/src/SIMPL/Source/SIMPLib/ITK/Testing/Cxx/itkTransformToDream3DTransformContainerTest.cpp:55:15:   required from here
/home/matt/src/SIMPL/Source/SIMPLib/ITK/itkTransformToDream3DTransformContainer.hxx:85:1: error: ‘void itk::TransformToDream3DTransformContainer<ITKTransformType>::VerifyPreconditions() [with ITKTransformType = itk::AffineTransform<float, 3>]’ marked ‘override’, but does not override
 TransformToDream3DTransformContainer<ITKTransformType>
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/matt/src/SIMPL/Source/SIMPLib/ITK/itkTransformToDream3DITransformContainer.h:81:0,
                 from /home/matt/src/SIMPL/Source/SIMPLib/ITK/Testing/Cxx/itkTransformToDream3DITransformContainerTest.cpp:33,
                 from SIMPLView/SIMPL/SIMPLib/Testing/SIMPLUnitTest.cpp:43:
/home/matt/src/SIMPL/Source/SIMPLib/ITK/itkTransformToDream3DITransformContainer.h: In instantiation of ‘class itk::TransformToDream3DITransformContainer<float, 3>’:
/home/matt/src/SIMPL/Source/SIMPLib/ITK/Testing/Cxx/itkTransformToDream3DITransformContainerTest.cpp:54:15:   required from here
/home/matt/src/SIMPL/Source/SIMPLib/ITK/itkTransformToDream3DITransformContainer.hxx:86:1: error: ‘void itk::TransformToDream3DITransformContainer<TParametersValueType, CompositeNDimensions>::VerifyPreconditions() [with TParametersValueType = float; unsigned int CompositeNDimensions = 3]’ marked ‘override’, but does not override
 TransformToDream3DITransformContainer<TParametersValueType, CompositeNDimensions>
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[395/1105] Building CXX object SIMPLView/SIMPL/SVWidgetsLib/CMakeFiles/SVWidgetsLib.dir/FilterParameterWidgets/ReadASCIIDataWidget.cpp.o

```

**Comment by joeykleingers** [comment]
This build error is occurring because we haven't updated SIMPLib to support ITK 5 yet.  We do, however, have the updates sitting on a branch called OSD_GUI on the SIMPL repo.  I just created a branch on the MultiscaleFusion repo called OSD_GUI as well.  If you guys could set this pull request to go into the OSD_GUI branch instead of develop, then we can continue development on the OSD_GUI branches until we are ready to merge everything into develop.

**Comment by dzenanz** [comment]
@thewtex this is supposed to build against a slightly [modified](https://github.com/dzenanz/DREAM3DSuperbuild/commit/f309b6718a6bee0fdaeecd05f20141d88e3cdfe8) super-build.

**Pull request closed without merging**